src/chainrules.jl

```diff
@@ -0,0 +1,40 @@
+@non_differentiable validate_woodbury_arguments(A, D, S)
+
+function ChainRulesCore.rrule(
+    ::typeof(*), A::Real, B::WoodburyPDMat{T, TA, TD, TS}
+) where {T, TA, TD, TS}
+    project_A = ProjectTo(A)
+    project_B = ProjectTo(B)
+    function times_pullback(ȳ::AbstractMatrix)
+        Ȳ = unthunk(ȳ)
+        Ā = dot(Ȳ, B)
+        B̄ = A' * Ȳ
+        return (
+           NoTangent(),
+           @thunk(project_A(Ā')),
+           @thunk(project_B(B̄)),
+        )
+    end
```

Comment:
```suggestion
    function times_pullback(ȳ::AbstractMatrix)
        Ā = @thunk(project_A(dot(Ȳ, B)'))
        B̄ = @thunk(project_B(A' * Ȳ))
        return (NoTangent(), Ā, B̄)
    end
```
we already know it is not a thunk from the pullback argument type